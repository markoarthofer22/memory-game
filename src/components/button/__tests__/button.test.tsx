import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import Button from '../index';

describe('Button component tests', () => {
  it('First render with all props passed', () => {
    const component = renderer.create(<Button title="Button title" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render a button with the tile "Button title"', () => {
    render(<Button title="Button title" />);
    const button = screen.getByText(/title/i);

    expect(button).toHaveTextContent('Button title');
  });

  it('should be disabled if disabled prop is passed', () => {
    render(<Button disabled title="Button title" />);
    const button = screen.getByText(/title/i);

    expect(button).toBeDisabled();
  });

  it('should be disabled when clicked', async () => {
    let disabled = false;
    const setDisabled = jest.fn(() => (disabled = !disabled));

    const { rerender } = render(
      <Button
        disabled={disabled}
        title="Button title"
        onClick={() => {
          setDisabled();
        }}
      />
    );
    const button = screen.getByText(/title/i);

    expect(button).not.toBeDisabled();

    await user.click(button);

    rerender(
      <Button
        disabled={disabled}
        title="Button title"
        onClick={() => {
          setDisabled();
        }}
      />
    );

    expect(button).toBeDisabled();
  });
});
