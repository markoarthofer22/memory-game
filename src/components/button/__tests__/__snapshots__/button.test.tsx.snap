// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests First render with all props passed 1`] = `
<button
  className=" "
  disabled={false}
  title="Button title"
  type="button"
>
   
  Button title
</button>
`;
